CppSamples/Search/FindPlace/FindPlace.cpp

```diff
@@ -157,17 +158,16 @@ void FindPlace::startLocationPermission()
   QLocationPermission locationPermission{};
   locationPermission.setAccuracy(QLocationPermission::Accuracy::Precise);
   locationPermission.setAvailability(QLocationPermission::Availability::WhenInUse);
-  switch (qApp->checkPermission(locationPermission))
+
+  qApp->requestPermission(locationPermission, [this, locationPermission](const QPermission& permission)
   {
-  case Qt::PermissionStatus::Undetermined:
-    qApp->requestPermission(locationPermission, this, &FindPlace::startLocationPermission);
-    return;
-  case Qt::PermissionStatus::Granted:
-    return;
-  case Qt::PermissionStatus::Denied:
-    emit locationPermissionDenied();
-    return;
-  }
+    Q_UNUSED(permission)
+    if (qApp->checkPermission(locationPermission) == Qt::PermissionStatus::Denied)
+    {
+      emit locationPermissionDenied();
+    }
+    connectSignals();
```

Comment:
If location permissions are denied, should you still call `connectSignals()`?

Reply:
When the user searches for a location by name instead of using their current location, a callout should be displayed upon a mouse click which is also written in `connectSignals`